src/Corax/Analyzers/Analyzer.cs

```diff
@@ -319,8 +319,54 @@ static void RunUtf8WithConversion(Analyzer analyzer, ReadOnlySpan<byte> source,
                 characters = Encoding.UTF8.GetBytes(charBuffer, output);
                 output = output.Slice(0, characters);
 
-                // TODO: Adjust the tokens considering that we may be changing from 4 bytes to less and from 2 to 1.
-                //       This will cause all non-ascii test to fail. We know that, we dont care for now. 
+                // Adjust the tokens considering that we may be changing from 4 bytes to less and from 2 to 1.
+                var outputPtr = (byte*) Unsafe.AsPointer(ref MemoryMarshal.GetReference(output));
+                var endPtr = outputPtr + output.Length;
+
+                uint excessBytes = 0;
+                uint totalExcessBytes = 0;
+                int currentToken = 0;
+                int processedChars = 0;
+                while (outputPtr != endPtr)
+                {
+                    var @byte = *outputPtr;
+                    if (@byte <= 0x7F)
+                    {
+                        /* 1 byte sequence: U+0000..U+007F */
+                        // Nothing to do here. 
+                    }
+                    else if (0xC2 <= @byte && @byte <= 0xDF)
```

Comment:
Why not `@byte & 0b1100_0000 == 0b1100_0000` - Seems much clearer to understand

Reply:
See: https://sharplab.io/#v2:C4LghgzgtgPgAgJgIwFgBQcDMACR2DC2A3utmbjgJYB2w2ArsABwD6ElAXgKYtIAUAIQCewLtgBGASmKlyc7JQBm2PuOwAeALzYADAA8A7ADFJs+eYsk0Fm7bgHsSANxnbNgL6u35rgBsIYkoq+vgIGtpqAGSREuG6egAiJl7eclapqfbYCC7WGeaeeflyfgEKynz6AKI6cVExalrxVclFxWTp7RZZmLld5IX9ZKWBFfpGtU31sU3jACymbe2dQ+RZc31Dg90OALTOrttkrlgKtAzMbJw8CIIiYlIyS+RKfKrYMTriSDq/LL+/aSabRfH5/AE6RY2FZ2BwHZ5kI7yEblN5RXTfMH/AFAkGY8E4lIdInyLI5ElIkr+UZoj4YpAMnTYwHYYH0xnMyEkmE2HqbAoklGvd6fTEMzm49niiFQywkuTrfnySlrPbwuSFdxAA==

Reply:
Applied in my PR. https://github.com/ravendb/ravendb/pull/13230

Reply:
Not only much clearer, the assembler is better.